app/src/main/java/com/android/streetworkapp/model/preferences/PreferencesRepositoryDataStore.kt

```diff
@@ -0,0 +1,161 @@
+package com.android.streetworkapp.model.preferences
+
+import android.content.Context
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import java.io.IOException
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+
+class PreferencesRepositoryDataStore(context: Context) : PreferencesRepository {
+
+  private val Context.dataStore: DataStore<Preferences> by
+      preferencesDataStore(name = "preferences")
+  private val dataStore = context.dataStore
+
+  companion object {
+    val IS_LOGGED_IN = booleanPreferencesKey("is_logged_in")
+    val SAVED_UID = stringPreferencesKey("saved_uid")
+    val SAVED_NAME = stringPreferencesKey("saved_name")
+    val SAVED_SCORE = intPreferencesKey("saved_score")
+
+    private const val IO_ERR = "An IO exception occurred, emitting empty preferences."
```

Comment:
Suggestion: you could add the `DataStoreManager` as a constant since this tag is used multiple times

Reply:
Indeed, it's now implemented.